test/transactWriteItem.js

```diff
@@ -0,0 +1,1281 @@
+      assertValidation({ TransactItems: [ { Put: {} } ] },
+        '1 validation error detected: ' +
+                'Value null at \'transactItems.1.member.put.item\' failed to satisfy constraint: ' +
+                'Member must not be null', done)
+    })
+
+
+    it('should return ValidationException for invalid delete request in TransactItems', function (done) {
+      assertValidation({ TransactItems: [ { Delete: {} } ] },
+        '1 validation error detected: ' +
+                'Value null at \'transactItems.1.member.delete.key\' failed to satisfy constraint: ' +
+                'Member must not be null', done)
+    })
+
+    it('should return ValidationException for invalid ConditionCheck request in TransactItems', function (done) {
+      assertValidation({ TransactItems: [ { ConditionCheck: {} } ] },
+        '1 validation error detected: ' +
+                'Value null at \'transactItems.1.member.conditionCheck.key\' failed to satisfy constraint: ' +
+                'Member must not be null', done)
+    })
+
+    it('should return ValidationException for invalid metadata and missing requests', function (done) {
+      assertValidation({ TransactItems: [], ReturnConsumedCapacity: 'hi', ReturnItemCollectionMetrics: 'hi' }, [
+        'Value \'hi\' at \'returnConsumedCapacity\' failed to satisfy constraint: ' +
+                'Member must satisfy enum value set: [INDEXES, TOTAL, NONE]',
+        'Value \'hi\' at \'returnItemCollectionMetrics\' failed to satisfy constraint: ' +
+                'Member must satisfy enum value set: [SIZE, NONE]',
+        'Value \'[]\' at \'transactItems\' failed to satisfy constraint: ' +
+                'Member must have length greater than or equal to 1',
+      ], done)
+    })
+
+    it('should return ValidationException for incorrect attributes', function (done) {
+      assertValidation({
+        TransactItems: [ { Put: {}, Delete: {} } ],
+        ReturnConsumedCapacity: 'hi', ReturnItemCollectionMetrics: 'hi'
+      }, [
+        'Value \'hi\' at \'returnConsumedCapacity\' failed to satisfy constraint: ' +
+                'Member must satisfy enum value set: [INDEXES, TOTAL, NONE]',
+        'Value \'hi\' at \'returnItemCollectionMetrics\' failed to satisfy constraint: ' +
+                'Member must satisfy enum value set: [SIZE, NONE]',
+        'Value null at \'transactItems.1.member.delete.key\' failed to satisfy constraint: ' +
+                'Member must not be null',
+        'Value null at \'transactItems.1.member.put.item\' failed to satisfy constraint: ' +
+                'Member must not be null',
+      ], done)
+    })
+
+    it('should return ValidationException when writing more than 25 items', function (done) {
+      var requests = [], i
+      for (i = 0; i < 26; i++) {
+        requests.push(i % 2 ? { Delete: { Key: { a: { S: String(i) } } } } : { Put: { Item: { a: { S: String(i) } } } })
+      }
```

Comment:
Maybe something like this:

```suggestion
      const requests = new Array(26).fill(null).map(() => ({...}))
```